pkg/cache/sql/db/client.go

```diff
@@ -343,3 +353,14 @@ func (c *Client) NewConnection() error {
 	c.conn = sqlDB
 	return nil
 }
+
+// This acts like "touch" for non-existent files...setting the correct
+// permissions.
+func touchFile(filename string, perms fs.FileMode) error {
+	f, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE, perms)
+	if err != nil {
+		return err
+	}
+
```

Comment:
If the file already existed, this won't enforce the target file mode. We could add a `chmod` here to that effect. That way, we always ensure the correct file mode is in place. Thoughts?

Reply:
I am not sure we _need_ to just now?

Definitely could be documented, but the only place it's called from is just after it's [deleted](https://github.com/rancher/lasso/blob/3c49d3df84e2025d5f879209b6254edca18c66f1/pkg/cache/sql/db/client.go#L322) ?

It could be a generic "test for file, create if doesn't exist with right perms otherwise, os.Chmod" function? Admittedly, that's racy, so we could opt to do that just after we create it...

Happy to add it in case it's used later for something else.